### ISSUE TITLE
Make amp-options an autoloaded option

### ISSUE THREAD

**Issue by westonruter** [ampproject/amp-wp#3287]
On accessing the frontend, there is a persistent query for the `amp-options` option:

<img width="967" alt="Screen Shot 2019-09-17 at 22 25 04" src="https://user-images.githubusercontent.com/134745/65114121-7608f400-d99a-11e9-97e2-e3f927938814.png">

Given that this option is always queried with each request, the option should be made autoloaded (part of `alloptions`).

**Comment by MackenzieHartung** [comment]
Note: For now, Analytics will be decoupled from the other AMP options stored, and remain not autoloaded.

**Comment by westonruter** [comment]
We should make sure that any related WordPress core issues related to autoloaded options don't cause headaches here. I recall at a time that there was a bug in core where multiple updates to an autoloaded option could cause some updates to be lost. If this issue is still current, then it will be a problem for the AMP plugin since `amp-options` stores multiple settings that could receive more than one update in a request.

**Comment by bozzmedia** [comment]
Stopping by on this ticket after AAA Optimizer reported 1250593276 calls to amp-options over the course of an hour on a fairly busy site. Is it safe to make this option autoloaded?

**Comment by westonruter** [comment]
@bozzmedia Yes, it should be. The size of the stored option shouldn't be overly large. Here is the default value of the option:

https://github.com/ampproject/amp-wp/blob/5debbdfc2486aa241e362b4730bf447a1f701359/includes/options/class-amp-options-manager.php#L30-L42
